registry/client.go

```diff
@@ -773,6 +779,51 @@ func (c *client) ForEachAssignedScanner(agentID string, handler func(s *Scanner)
 	return nil
 }
 
+func (c *client) IndexOfAssignedScannerByChain(agentID, scannerID string, chainID *big.Int) (*big.Int, error) {
+	opts, err := c.getOpts()
+	if err != nil {
+		return nil, err
+	}
+	aID := utils.AgentHexToBigInt(agentID)
+	sID := utils.ScannerIDHexToBigInt(scannerID)
+	length, err := c.Contracts().Dispatch.NumScannersFor(opts, aID)
+	if err != nil {
+		return nil, err
+	}
+
+	var idxByChain int64
+	for i := int64(0); i < length.Int64(); i++ {
+		idx := big.NewInt(i)
+		scn, err := c.Contracts().Dispatch.ScannerRefAt(opts, aID, idx)
```

Comment:
Getting the contracts is gated by a mutex and requires copying the struct. It could be a little better to define `contracts := c.Contracts()` once on the top and reuse `contracts` in the two lines in this method

Reply:
Fixed with f61c80dea61b572b1a1d237f19b48e383339d7f3